File: client/src/app/shared/models/order.ts
export interface OrderItem {
  productId: number;
  productName: string;
  imageUrl: string;
  price: number;
  quantity: number;
  discount: number;
}

export interface OrderToCreate {
    basketId: string; 
}

export interface Order {
  id: number
  buyerEmail: string
  orderdate: string
  orderItems: OrderItem[]
  total: number
}

File: Core/Entities/OrderAggregate/Order.cs
namespace Core.Entities.OrderAggregate
{
    public class Order : BaseEntity
    {
        public Order()
        {
        }

        public Order(IReadOnlyList<OrderItem> orderItems, string buyerEmail, decimal total)
        {
            BuyerEmail = buyerEmail;
            OrderItems = orderItems;
            Total = total;
        }

        public string BuyerEmail { get; set; }
        public DateTime Orderdate { get; set; } = DateTime.UtcNow;
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public decimal Total { get; set; }
    }
}